File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution {

    /**
     * @param ListNode $head
     * @return ListNode
     */
    function swapPairs($head) {
        $dummy = new ListNode(0, $head);
        $prev = $dummy;
        $cur = $dummy->next;
        while($cur && $cur->next){
            $next = $cur->next;
            
            $prev->next = $next;
            $cur->next = $next->next;
            $next->next = $cur;
            
            $prev = $cur;
            $cur = $cur->next;
        }
        return $dummy->next;
    }
}

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swapPairs(head *ListNode) *ListNode {
    dummy := &ListNode{Next: head}
    prev := dummy
    cur := dummy.Next
    
    for cur != nil && cur.Next != nil {
        next := cur.Next
        
        prev.Next = next
        cur.Next = next.Next
        next.Next = cur
        
        prev = cur
        cur = cur.Next
    }
    
    return dummy.Next
}